### PR TITLE
Add Artwork to Group

### LINKED ISSUE #126: Add artwork to groups
Allow users to selectively choose what artworks they add to their groups.

### PR DESCRIPTION
**Pull request by Waidhoferj**
Fixes #126 
Users can now add artworks to their groups on the `Artwork` page.

## Testing
1. Find an artwork that is in your personal portfolio which is not in the group portfolio. You can edit the initialization of your portfolio in `database.py` to include works that are not in your group's portfolio.
2. Click on the unique artwork in your personal portfolio.
3. On the Artwork page, scroll down and click on the group icon button with the blue background.
4. On the popup drawer, choose the group that does not have this artwork.
5. Go to the group and verify that the artwork was added.
6. Also test that adding an artwork that is already in the group portfolio does not create a duplicate.
## Other Changes
- Fixed the backend function that generates images. Previously picked up on files like `.DS_Store` which led to missing images on the frontend.
- Added an options section to the `GroupPage` and colored the "Leave Group" button red.
- Added general `Drawer` component that allows you to put content in a modal that slides up from the bottom of the screen.

### CODE REVIEW

File: frontend/src/components/Drawer/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import "./Drawer.scss";
+import { motion } from "framer-motion";
+interface DrawerProps {
+  title?: string;
+  children?: React.ReactNode;
+  onClose: () => void;
+}
+
+export default function Drawer(props: DrawerProps) {
```
Comment (Braden50) [inline frontend/src/components/Drawer/Drawer.tsx:9]: Clever decision of component. I really like how it shows up on the frontend!

**Review by kyleshare — APPROVED** [review]
Works great. Looking at the set up I'm thinking it may be difficult to remove artworks that a user adds to the group when the user leaves that group. Maybe we don't need to implement that feature.

File: frontend/src/pages/Artwork/gql.ts [hunk 2]
```diff
@@ -56,6 +56,54 @@ export const GET_ARTWORK = gql`
...
+  return (
+    data?.users.edges?.[0].node.groups.edges.map((e: any) => ({
+      name: e.node.name,
+      id: e.node.id,
+    })) || []
+  );
+}
+
+export const ADD_ARTWORK_TO_GROUP = gql`
+  mutation addArtworkToGroup($artworkId: String!, $groupId: String!) {
```
Comment (Braden50) [inline frontend/src/pages/Artwork/gql.ts:98]: Good work on creating the whole mutation here. The more mutations we have on the frontend the easier it gets to use our backend!

File: frontend/src/pages/Groups/GroupPage/GroupPage.tsx [hunk 3]
```diff
@@ -153,6 +150,14 @@ function GroupHub({ group }: GroupHubProps) {
             />
           </div>
         </section>
+        <section>
+          <h2>Options</h2>
+          <div className="actions">
+            <button className="danger" onClick={leaveGroup}>
```
Comment (Braden50) [inline frontend/src/pages/Groups/GroupPage/GroupPage.tsx:156]: Love the new position and styling of the button. Nice job!

**Comment by gwholland3** [comment]
Looks great! One thought I had is maybe we should have the Choose Group drawer not display groups that already contain the selected artwork, to avoid user confusion

**Review by Braden50 — APPROVED** [review]
All your additions are clean and work very well. No negative comments, nice job!

**Comment by Waidhoferj** [comment]
@kyleshare Agreed, we would probably have to rework how the artwork is stored so that it has reference back to the user. But in this case, its probably fine to keep artworks in the portfolio from past users.

File: backend/testing.py [hunk 4]
```diff
@@ -439,7 +439,7 @@ def get_sample_encoded_art_image(filepath="") -> str:
     if filepath == "":
         # select random image if not supplied with a path
         path = "./assets/sample-artworks"
-        image_paths = os.listdir(path)
+        image_paths = [img for img in os.listdir(path) if img.endswith(".jpg")]
```
Comment (gwholland3) [inline backend/testing.py:442]: Is this to address what you were mentioning about `.DS_Store`?

File: frontend/src/pages/Groups/GroupPage/GroupPage.tsx [hunk 5]
```diff
@@ -77,19 +77,18 @@ function GroupHub({ group }: GroupHubProps) {
   const { id } = useParams<{ id: string }>();
   const { goBack, push } = useHistory();
   const [leaveGroupMutation] = useMutation(LEAVE_GROUP);
-  const { profile: user} = useProfileInfo();
+  const { profile: user } = useProfileInfo();
```
Comment (gwholland3) [inline frontend/src/pages/Groups/GroupPage/GroupPage.tsx:80]: Is it convention in JS to put spaces in between curly braces and their contents when dereferencing?

File: frontend/src/pages/Groups/GroupPage/GroupPage.tsx [hunk 6]
```diff
@@ -77,19 +77,18 @@ function GroupHub({ group }: GroupHubProps) {
...
-    let resp = await leaveGroupMutation({ variables: payload }); 
+      group: id,
+    };
+    let resp = await leaveGroupMutation({ variables: payload });
 
     if (resp["data"]["leaveGroup"]["success"]) {
       push("/groups");
-    }
-    else {
+    } else {
```
Comment (gwholland3) [inline frontend/src/pages/Groups/GroupPage/GroupPage.tsx:91]: I like the syntax cleanup!

File: frontend/src/pages/Artwork/Artwork.tsx [hunk 7]
```diff
@@ -131,3 +150,47 @@ export default function Artwork() {
     </Switch>
   );
 }
+
+interface GroupDrawerProps {
```
Comment (kyleshare) [inline frontend/src/pages/Artwork/Artwork.tsx:154]: I haven't used any interfaced in my components, this is smart.

File: frontend/src/pages/Artwork/Artwork.tsx [hunk 8]
```diff
@@ -112,8 +120,19 @@ export default function Artwork() {
               <button>
                 <Camera />
               </button>
+              <button onClick={() => setShowGroupDrawer(true)}>
+                <Users />
+              </button>
             </div>
           </div>
+          <AnimatePresence>
```
Comment (kyleshare) [inline frontend/src/pages/Artwork/Artwork.tsx:128]: I like this component, I haven't seen it used before

File: backend/testing.py (hunk 4 above)
Comment (Waidhoferj) [reply backend/testing.py:442]: Yeah and any other files that aren't images. Long term if we add other image types, I'll replace the `endswith` with a regex `/png|jpe*g|webp/` to cover all our img suffixes

File: frontend/src/pages/Groups/GroupPage/GroupPage.tsx (hunk 5 above)
Comment (Waidhoferj) [reply frontend/src/pages/Groups/GroupPage/GroupPage.tsx:80]: Huh I never noticed that. Probably, I just let the Prettier gods handle the formatting 🦾

File: frontend/src/pages/Artwork/Artwork.tsx (hunk 7 above)
Comment (Waidhoferj) [reply frontend/src/pages/Artwork/Artwork.tsx:154]: Yeah! It gives you type completion and is a bit looser than stating `type GroupDrawerProps`. Definitely helps communicate intention as well.

File: frontend/src/pages/Artwork/Artwork.tsx (hunk 8 above)
Comment (Waidhoferj) [reply frontend/src/pages/Artwork/Artwork.tsx:128]: It's responsible for controlling the drawer entrance and exit animation. If you're interested here's the [link](https://www.framer.com/api/motion/animate-presence/) to the docs
